pandas/core/groupby/groupby.py

```diff
@@ -2015,10 +2015,16 @@ def _transform(self, func, *args, engine=None, engine_kwargs=None, **kwargs):
             with com.temp_setattr(self, "as_index", True):
                 # GH#49834 - result needs groups in the index for
                 # _wrap_transform_fast_result
-                if engine is not None:
-                    kwargs["engine"] = engine
-                    kwargs["engine_kwargs"] = engine_kwargs
-                result = getattr(self, func)(*args, **kwargs)
+                if func in ["idxmin", "idxmax"]:
+                    func = cast(Literal["idxmin", "idxmax"], func)
+                    result = self._idxmax_idxmin(
+                        func, ignore_unobserved=True, axis=self.axis, **kwargs
```

Comment:
We definitely don't need *args here right? Seems like something could get discarded compared to the other branch

Reply:
Ack - thanks! Fixed and test added.